src/main/java/uk/ac/ebi/embl/converter/gff3/reader/GFF3FileReader.java

```diff
@@ -0,0 +1,176 @@
+                String accession = m.group("accession");
+                long start = Long.parseLong(m.group("start"));
+                long end = Long.parseLong(m.group("end"));
+                // TODO: Validation no multiple sequence accession
+                // TODO: Validation Sequence accession is required!
+                GFF3Directives.GFF3SequenceRegion sequenceAccession =
+                        new GFF3Directives.GFF3SequenceRegion(accession, start, end);
+                directives.add(sequenceAccession);
+                continue;
+            }
+            m = DIRECTIVE_SPECIES.matcher(line);
+            if (m.matches()) {
+                GFF3Directives.GFF3Species species = new GFF3Directives.GFF3Species(m.group("species"));
+                // TODO: Validation no multiple species
+                directives.add(species);
+                continue;
+            }
+            m = COMMENT.matcher(line);
+            if (m.matches()) {
+                // Skip comment
+                continue;
+            }
+            m = GFF3_FEATURE.matcher(line);
+            if (m.matches()) {
+                String accession = m.group("accession");
+                String source = m.group("source");
+                String name = m.group("name");
+                long start = Long.parseLong(m.group("start"));
+                long end = Long.parseLong(m.group("end"));
+                String score = m.group("score");
+                String strand = m.group("strand");
+                String phase = m.group("phase");
+                String attributes = m.group("attributes");
+                Map<String, String> attributesMap = attributesFromString(attributes);
+                Optional<String> id =
+                        attributesMap.containsKey("ID") ? Optional.of(attributesMap.get("ID")) : Optional.empty();
+                Optional<String> parentId = attributesMap.containsKey("Parent")
+                        ? Optional.of(attributesMap.get("Parent"))
+                        : Optional.empty();
+
+                GFF3Feature feature = new GFF3Feature(
+                        id, parentId, accession, source, name, start, end, score, strand, phase, attributesMap);
+                id.ifPresent(s -> featureIdx.put(s, feature));
+
+                if (parentId.isPresent()) {
+                    if (featureIdx.containsKey(parentId.get())) {
+                        featureIdx.get(parentId.get()).addChild(feature);
+                    }
+                    // TODO: Validation check, no parent found
+                } else if (attributesMap.containsKey("gene")) {
+                    String gene = attributesMap.get("gene");
+                    geneMap.putIfAbsent(gene, new ArrayList<>());
+                    List<GFF3Feature> geneFeatures = geneMap.get(gene);
+                    geneFeatures.add(feature);
+                } else {
+                    nonGeneFeatures.add(feature);
+                }
```

Comment:
Question: Do we need this in case of GFF3-FF conversion? Can't we just add a constructor to write directives and all the features to FF features?

` GFF3Annotation(new GFF3Directives(directives), features);`